contracts/DistributionCreator.sol

```diff
@@ -118,29 +115,50 @@ contract DistributionCreator is UUPSHelper, ReentrancyGuardUpgradeable {
     /// @notice List of all reward tokens that have at some point been accepted
     address[] public rewardTokens;
 
-    uint256[36] private __gap;
+    /// @notice List of all rewards ever distributed or to be distributed in the contract
+    /// @dev An attacker could try to populate this list. It shouldn't be an issue as only view functions
+    /// iterate on it
+    CampaignParameters[] public campaignList;
+
+    /// @notice Maps a campaignId to the ID of the campaign in the campaign list + 1
+    mapping(bytes32 => uint256) internal _campaignLookup;
+
+    /// @notice Maps a campaign type to the fees for this specific campaign
+    mapping(uint32 => uint256) public campaignSpecificFees;
 
-    // =================================== EVENTS ==================================
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        EVENTS                                                      
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
 
     event DistributorUpdated(address indexed _distributor);
     event FeeRebateUpdated(address indexed user, uint256 userFeeRebate);
     event FeeRecipientUpdated(address indexed _feeRecipient);
     event FeesSet(uint256 _fees);
+    event CampaignSpecificFeesSet(uint32 campaignType, uint256 _fees);
     event MessageUpdated(bytes32 _messageHash);
+    event NewCampaign(CampaignParameters campaign);
     event NewDistribution(DistributionParameters distribution, address indexed sender);
     event RewardTokenMinimumAmountUpdated(address indexed token, uint256 amount);
     event TokenWhitelistToggled(address indexed token, uint256 toggleStatus);
     event UserSigned(bytes32 messageHash, address indexed user);
     event UserSigningWhitelistToggled(address indexed user, uint256 toggleStatus);
 
-    // ================================= MODIFIERS =================================
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                       MODIFIERS                                                    
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
 
     /// @notice Checks whether the `msg.sender` has the governor role or the guardian role
     modifier onlyGovernorOrGuardian() {
```

Comment:
What is the dif between `onlyGovernor` and `onlyGovernorUpgrader` as they should behave the same (failing if core is the null address)

Reply:
It's nearly the same thing indeed. It dates back from before this upgrade though